### DISCUSSION TITLE
URL to directly launch a new codespace?

### DISCUSSION THREAD

**Discussion by aaronsteers** [community/community#39208] [Codespaces] [answered]
### Select Topic Area

Question

### Body

I want to provide our end users a link that directly launches a new codespace from a pre-coded repo and configset. Is there a way to do something like the below, but without the need to click "Create codespace" after clicking the URL.

[`https://github.com/codespaces/new?machine=basicLinux32gb&repo=556945668&ref=main&devcontainer_path=.devcontainer%2Fdevcontainer.json&location=WestUs2`](https://github.com/codespaces/new?machine=basicLinux32gb&repo=556945668&ref=main&devcontainer_path=.devcontainer%2Fdevcontainer.json&location=WestUs2)

> ![image](https://user-images.githubusercontent.com/18150651/202061489-3a26c409-90cf-4023-a14a-a07408603d48.png)

---------------------

And same question, part 2, would be: can I do the same for template repos? Specifically, I want to provide a link that replaces having to find and click this button:

> ![image](https://user-images.githubusercontent.com/18150651/202061553-062a7966-0cbb-44e0-8e5a-a5675c7e00ce.png)

**Comment by adnamalin** [comment, marked as answer]
👋 Hi there!

For templates, we have a similar deep link experience. You can create one like: 

`https://github.com/codespaces/new?template_repository=<REPO_OWNER>/<REPO_NAME>` 

Here's an example: https://github.com/codespaces/new?template_repository=github/codespaces-react

For both creating a codespace and creating a template deep links, we do not support creating on click because users could unknowingly begin using codespace usage (free or paid) upon clicking. We want this action to be explicit for users.

**Reply by aaronsteers** [reply]
Hi, @adnamalin - thanks very much for your answer. The template version of this (`template_repository=...`) is exactly what I was looking for and looks like it will meet our needs in this regard. The nice rendering of the repo description below the "Use this template" also gives us a space to prepare the user for what they are about to launch. 👍

**Comment by LoneRifle** [comment]
I have a slightly different use case, which may be a feature request. 

We (@opengovsg) maintain a template repository, for a [starter kit](https://github.com/opengovsg/starter-kit) similar to the [T3 stack](https://github.com/t3-oss/create-t3-app). Our [instructions to getting started](https://github.com/opengovsg/starter-kit/blob/develop/docs/guides/getting-started-1/README.md) deliberately guides the user to set up their own repository with Vercel's workflow, so that they start out already having the ability to rapidly deploy their product and any changes to their userbase.

However, as a result, while we want to have a button for 1-click creation of Codespaces, we cannot use the approaches above, since want them to create their Codespace from the repo they previously created, not by creating a new repo from the template.

What we would like would be the ability to create a Codespace using a single, universal URL. That is, given, say, https://github.com/codespaces/from-referrer, a Codespace will be created using the repo implied by the page that the user clicked from. This is similar to what was proposed and subsequently implemented by Gitpod in gitpod-io/gitpod#78.

The advantage of this is that a README containing a Codespace creation button in a template repo would still work as intended, even after a new repo is created from the template and the button (and its URL) remains the same.
